### PR TITLE
SLING-8283 provide non render blocking version of Sling.getSessionInf…

### PR DESCRIPTION
**Pull request by reusr1**
…o and Sling.httpGet

### CODE REVIEW

**Comment by rombert** [comment]
A SonarQube report for the changes added _only by this pull request_ was generated. Please review it at https://builds.apache.org/job/Sling/job/sling-org-apache-sling-servlets-post/job/PR-4/1/artifact/target/sonar/issues-report/issues-report-light.html

File: src/main/resources/system/sling.js [hunk 1]
```diff
@@ -209,7 +217,24 @@ var Sling = null;
...
     }
-    
+
+    /** Get "session info" from repository. Mainly answers the question: "Who am I"
+     *  and "Which workspace am I logged into. Async version of getSessionInfo
+     * @param {Function} callback function, will get an An Object tree as first argument
+     *        containing the session information, null if server status <> 200
+     */
+    Sling.getSessionInfoAsync = function(callback) {
+        var res=Sling.httpGet(Sling.baseurl+"/system/sling/info.sessionInfo.json",
```
Comment (rombert) [inline src/main/resources/system/sling.js]: Nitpick - can you please write it as `var res = Sling...` ?

Also, do you have some testing code for the starter where you use this?

**Review by rombert — CHANGES_REQUESTED** [review]
LGTM, thanks @reusr1 ! Can you please address a formatting issue so I can merge this?

File: src/main/resources/system/sling.js (hunk 1 above)
Comment (rombert) [reply src/main/resources/system/sling.js]: To answer my second question, the minimally needed code to adapt the starter is

```
diff --git a/src/main/resources/content/content/starter/index.html b/src/main/resources/content/content/starter/index.html
index b07f627..6d93113 100755
--- a/src/main/resources/content/content/starter/index.html
+++ b/src/main/resources/content/content/starter/index.html
@@ -82,14 +82,15 @@
         </div>
         <script src="../system/sling.js"></script>
         <script>
-            var info = Sling.getSessionInfo();
-            if (info.userID !== 'anonymous') {
-                var login = document.querySelector('#Login');
-                var logout = document.querySelector('#Logout');
-                login.style.display = 'none';
-                logout.style.display = 'block';
-                logout.insertAdjacentHTML('beforeend', info.userID);
-            }
+            Sling.getSessionInfoAsync(function(info) {
+                if (info.userID !== 'anonymous') {
+                    var login = document.querySelector('#Login');
+                    var logout = document.querySelector('#Logout');
+                    login.style.display = 'none';
+                    logout.style.display = 'block';
+                    logout.insertAdjacentHTML('beforeend', info.userID);
+                }
+            });
         </script>
         <!-- Do not remove this comment, used for Starter integration tests -->
     </body>
```

Of course, there is a little flash when the change happens but that's not problem and definitely solvable in the starter, not here.

Comment (reusr1) [reply src/main/resources/system/sling.js]: changed the code style - however, that file is all over the place. And yes, that's where I was going with that change once the PR gets accepted

Comment (rombert) [reply src/main/resources/system/sling.js]: Thanks @reusr1 ! I agree the code style is inconsistent, I was only looking at your change though :-)

**Review by rombert — APPROVED** [review]

**Comment by rombert** [comment]
Thanks a lot for your contribution @reusr1 ! I squashed your changes and pushed them in a single commit.

**Comment by bdelacretaz** [comment]
Thanks for fixing this, that's code from...well, let's say _way back_ ;-)

**Pull request closed without merging**